chainerx_cc/chainerx/cuda/cuda_device/indexing.cu

```diff
@@ -173,8 +155,36 @@ __global__ void AddAtClipCudaKernel(
     }
 }
 
+Shape collapse_nd_array(const Array& a){
+
+    std::vector<int64_t> new_strides;
+    std::vector<int64_t> new_shape;
```

Comment:
Why not use `Shape` and `Strides`? They support `push_back`.

https://github.com/chainer/chainer/blob/6c93de466d6d9416c11ef3d0c9ab4eed66999ce4/chainerx_cc/chainerx/stack_vector.h#L158-L163